### PR TITLE
[12.0][FIX] membership_delegated_partner: changing delegated member

### LINKED ISSUE #78: [BUG] membership_delegated_partner: changing delegated member
Changing delegated member in invoice doesn't update correctly the membership lines.

Tested in V12, probably also occurs on previous versions.

Steps to reproduce:

- Enable group "Delegated partner in membership" for current user.
- Create a Membership Product with Duration that includes the date of today.
- Create an invoice with the above Membership Product. Add a delegated member. Save.
- A line is created in the Membership lines of the selected delegated member. OK.
- No line is present in the Membership lines of the customer. OK.
- Edit the invoice. Remove the delegated member. Save.

Result:
- The membership is NOT updated: no new line is created in the Membership lines of the customer. NOT OK.
- The membership is still present in the Membership lines of the delegated member. NOT OK.

Expected:
- The membership must be updated, a new line must be created in the members lines of the customer.
- No membership line must be present in the members lines of the delegated member.

### PR DESCRIPTION
**Pull request by astirpe**
Fixes https://github.com/OCA/vertical-association/issues/78

~This PR is also a proposal to use a different approach for calculating the membership lines.
The proposal is to use a compute method for the partner in the membership lines.~

### CODE REVIEW

**Review by chienandalu — CHANGES_REQUESTED** [review]
Looks good, although tests are failing...

File: membership_delegated_partner/models/membership_line.py [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
 
-from odoo import api, models
+from odoo import api, models, fields
 
 
 class MembershipLine(models.Model):
     _inherit = 'membership.membership_line'
 
+    partner = fields.Many2one(
+        compute='_compute_partner', store=True, readonly=False)
```
Comment (chienandalu) [inline membership_delegated_partner/models/membership_line.py]: Maybe this aproach is too 13ish for this version :sweat_smile:

Comment (astirpe) [reply membership_delegated_partner/models/membership_line.py]: Indeed in V13 doesn't break the tests :laughing: 
I'll take a look at it tomorrow...

**Comment by astirpe** [comment]
@chienandalu Fixed!

**Review by pedrobaeza — APPROVED** [review]

**Comment by astirpe** [comment]
@chienandalu would you update your review?

**Review by chienandalu — APPROVED** [review]
Thanks @astirpe :)

**Pull request merged**